meta-k8s-setup/recipes-k8s-configuration/k8s-configuration/files/k8s-configuration.service

```
@@ -0,0 +1,20 @@
+[Unit]
+Description=service that unpacks the configuration files
+
+After=config.mount
```

```suggestion
Before=sysinit.target
```
We don't want `After=config.mount`, but if we did `RequiresMountsFor=/config` is the way to do it.
We also want this service to start before "normal services" like `crio.service` and `kubelet.service`, so add ``Before=sysinit.target` (I'm unsure if that will work tho).

---

This will not work, try:
```suggestion
DefaultDependencies=no
Conflicts=shutdown.target
After=local-fs.target
Before=sysinit.target shutdown.target
```
or we could just 
```suggestion
Before=multi-user.target
```
but then it would be starting a bit later (but still early enough), hmm🤔

See https://www.freedesktop.org/software/systemd/man/bootup.html#System%20Manager%20Bootup for the boot process.

---

Well we need to make sure that the service is not started before the config partition is mounted, so how do we do that if not with: After=config.mount

---

We enable the service with the Yocto systemd class